### PR TITLE
mosquitto: new recipe

### PR DESCRIPTION
**Pull request by hoanga**
New recipe for mosquitto, a mqtt reference implementation

### CODE REVIEW

File: net-misc/mosquitto/mosquitto-1.6.9.recipe [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	cmd:gcc$secondaryArchPrefix
+	cmd:make
+	cmd:sed
+	cmd:xsltproc
+	cmd:pkg_config$secondaryArchPrefix
+	"
+
+BUILD()
+{
+	make
```
Comment (Begasus) [inline net-misc/mosquitto/mosquitto-1.6.9.recipe:50]: One space too many after "make" (Trailing whitespace)

File: net-misc/mosquitto/mosquitto-1.6.9.recipe [hunk 2]
```diff
@@ -0,0 +1,61 @@
+SUMMARY="An open source MQTT broker"
+DESCRIPTION="An open source MQTT brokerz"
```
Comment (extrowerk) [inline net-misc/mosquitto/mosquitto-1.6.9.recipe:2]: Should be different.

File: net-misc/mosquitto/mosquitto-1.6.9.recipe [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+HOMEPAGE="http://mosquitto.org"
+COPYRIGHT="2020 Eclipse Foundation."
+LICENSE="EDL"
+REVISION="1"
+SOURCE_URI="https://mosquitto.org/files/source/mosquitto-$portVersion.tar.gz"
+CHECKSUM_SHA256="412979b2db0a0020bd02fa64f0a0de9e7000b84462586e32b67f29bb1f6c1685"
+PATCHES="mosquitto-$portVersion.patch"
```
Comment (extrowerk) [inline net-misc/mosquitto/mosquitto-1.6.9.recipe:9]: Should be a patchset.

File: net-misc/mosquitto/mosquitto-1.6.9.recipe [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+BUILD()
+{
+	make 
+}
+
+INSTALL()
+{
+	mkdir -p $binDir
+	mkdir -p $docDir
+	#prefix=$prefix incdir=$includeDir mandir=$manDir make install
```
Comment (extrowerk) [inline net-misc/mosquitto/mosquitto-1.6.9.recipe:57]: Should be removed if unneeded.

**Comment by extrowerk** [comment]
Headers should go in a _devel subpackage.

File: net-misc/mosquitto/mosquitto-1.6.9.recipe [hunk 5]
```diff
@@ -30,6 +31,19 @@ REQUIRES="
...
+	mosquitto${secondaryArchSuffix}_devel = $portVersionCompat
+	devel:libmosquitto$secondaryArchSuffix = $libVersionCompat
+	"
+
+REQUIRES_devel="
+	mosquitto$secondaryArchSuffix == $portVersion base
+	"
+
+CONFLICTS_devel="
+	mosquitto${secondaryArchSuffix}_devel
```
Comment (extrowerk) [inline net-misc/mosquitto/mosquitto-1.6.9.recipe:44]: Umm.. why?

**Comment by hoanga** [comment]
Attempted to separate the headers.

Not sure if there is a suggested way to use a local licenses file to work with the build check

File: net-misc/mosquitto/mosquitto-1.6.9.recipe (hunk 2 above)
Comment (extrowerk) [reply net-misc/mosquitto/mosquitto-1.6.9.recipe:2]: Not really different....

Comment (extrowerk) [reply net-misc/mosquitto/mosquitto-1.6.9.recipe:2]: Would use this instead:
"Eclipse Mosquitto is an open source (EPL/EDL licensed) message broker that implements the MQTT protocol versions 5.0, 3.1.1 and 3.1. Mosquitto is lightweight and is suitable for use on all devices from low power single board computers to full servers.
The MQTT protocol provides a lightweight method of carrying out messaging using a publish/subscribe model. This makes it suitable for Internet of Things messaging such as with low power sensors or mobile devices such as phones, embedded computers or microcontrollers."

File: net-misc/mosquitto/mosquitto-1.6.9.recipe (hunk 1 above)
Comment (Begasus) [reply net-misc/mosquitto/mosquitto-1.6.9.recipe:64]: It should be "make" not "make " whitespace still there ...

File: net-misc/mosquitto/mosquitto-1.6.9.recipe (hunk 4 above)
Comment (Begasus) [reply net-misc/mosquitto/mosquitto-1.6.9.recipe:71]: Still not solved ...

**Comment by Begasus** [comment]
Too many new lines (didn't check if it actually builds), check the guidelines! https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines

**Comment by Begasus** [comment]
@hoanga any progress on this?

**Comment by kallisti5** [comment]
Oh.. I was just looking for an mqtt library for Haiku.  @hoanga any progress on this one?

**Comment by hoanga** [comment]
When I have more time I can attempt to pick this back up. Patches welcome, meanwhile :)

In the meantime, leaving a WIP package if there is any interest
[mosquitto.zip](https://github.com/haikuports/haikuports/files/4945160/mosquitto.zip)

File: net-misc/mosquitto/mosquitto-1.6.9.recipe (hunk 2 above)
Comment (Begasus) [reply net-misc/mosquitto/mosquitto-1.6.9.recipe]: Resolved :)

**Comment by Begasus** [comment]
@hoanga updated the recipe to the latest version and a cmake build, re-created a new patchset
@kallisti5 could you check if this works for you?

**Pull request merged**